File: tests/fixtures/csharp-simple/src/Payments/PaymentService.cs
using CSharpSimple.Utils;

namespace CSharpSimple.Payments
{
    public class PaymentService : IPaymentService
    {
        private readonly Logger _logger;

        public PaymentService(Logger logger)
        {
            _logger = logger;
        }

        public async Task<bool> ProcessPayment(decimal amount, string userId)
        {
            _logger.Info("Processing payment");
            this.ValidateAmount(amount);
            return true;
        }

        public async Task<bool> RefundPayment(string transactionId)
        {
            _logger.Info("Refunding");
            return true;
        }

        private bool ValidateAmount(decimal amount)
        {
            base.OnValidating(amount);
            return amount > 0;
        }

        public string DescribeStatus(PaymentStatus status)
        {
            switch (status)
            {
                case PaymentStatus.Pending:
                    return "Pending";
                case PaymentStatus.Completed:
                    return "Done";
                case PaymentStatus.Failed:
                    return "Error";
                default:
                    return "Unknown";
            }
        }
    }
}

File: tests/fixtures/java-simple/src/main/java/com/example/payments/PaymentService.java
package com.example.payments;

import com.example.utils.Logger;

/**
 * Handles payment processing.
 */
public class PaymentService implements IPaymentClient {

    private final Logger logger;
    private static int transactionCount;

    public PaymentService(Logger logger) {
        this.logger = logger;
        transactionCount = 0;
    }

    @Override
    public PaymentResult processPayment(String orderId, double amount) {
        logger.info("Processing payment for order: " + orderId);
        this.calculateFee(amount);
        transactionCount++;
        return new PaymentResult(orderId, true);
    }

    @Deprecated
    public synchronized void refund(String orderId) throws PaymentException {
        logger.info("Refunding order: " + orderId);
    }

    protected double calculateFee(double amount) {
        super.validate(amount);
        return amount * 0.03;
    }

    static int getTransactionCount() {
        return transactionCount;
    }

    public String describeResult(PaymentResult result) {
        switch (result) {
            case SUCCESS:
                return "Payment succeeded";
            case FAILED:
                return "Payment failed";
            case PENDING:
                return "Payment pending";
            default:
                return "Unknown";
        }
    }
}
